### DISCUSSION TITLE
Masowe pobieranie numerów przesyłek

### DISCUSSION THREAD

**Discussion by javaguru-pl** [allegro/allegro-api#8539] [Suggestions - rejected]
Z dokumentacji wynika, iż obecnie do pobierania numerów przesyłek dla zamówienia służy metoda:
https://api.{environment}/order/checkout-forms/{id}/shipments

Niestety ta metoda jest **baaardzo nieefektywna**, gdyż pozwala pobrać numery przesyłek **tylko dla pojedynczego zamówienia**. W przypadku aplikacji obsługującej setki tysięcy zamówień dziennie jest to bardzo złe rozwiązanie.

**Czy nie możecie rozważyć rozszerzenia tej metody o możliwość pobierania numerów przesyłek dla wielu zamówień na raz?**
Czyli podobnie jak w innych metodach byłyby dostępne parametry "from" (np. timestamp albo jakiś uuid), "offset", "limit" i sortowanie wyników (zamówień) licząc od tych z ostatnio dodanymi numerami przesyłek. W taki sposób można by błyskawicznie pobrać numery przesyłek dla tysięcy zamówień (dodatkowo tylko tych zamówień z ostatnio dodanymi numerami przesyłek)

P.S. Generalnie tworząc metody API warto mieć na uwadze - jak wygodna będzie dana metoda w przypadku dużego wolumenu danych (np. setek tysięcy). Metody typu "one request - one result" to prawdziwa zmora!

Z góry dzięki za odpowiedź.

**Comment by Lukasz-Zurek** [comment]
@javaguru-pl Poruszasz słuszną kwestię - chcemy, aby nasze API było optymalne i efektywne, to też zasób do pobierania numerów przesyłek dla wielu zamówień i możliwość filtrowania spełniłyby te oczekiwania. Nie mieliśmy jeszcze tego typu sugestii i tu oczywiście temat ten przekażę do odpowiedniego zespołu.

**Comment by javaguru-pl** [comment]
Kolejna sprawa, program Allegro Ceny. Aby sprawdzić czy dana aukcja jest w programie należy wywołać metodę:
https://developer.allegro.pl/documentation#operation/getAllegroPricesConsentForOffer
Metoda odpytuje POJEDYNCZĄ aukcję. Teraz mamy aplikację która obsługuje tysiące kont Allegro, łącznie dziesiątki milionów aukcji. Proszę mi powiedzieć w jaki sposób mamy wyświetlać aktualny stan aukcji pod względem programu Allegro Ceny (tzn. czy aukcja jest w programie czy też nie). Musielibyśmy codziennie odpytać kilkadziesiąt milionów aukcji. Jest to niewykonalne. Takie informacje powinny być zwracane w dzienniku w oraz w standardowych metodach sale/product-offers.
Scenariusz jest bardzo prosty:
1. Sprzedawca autoryzuje swoje konto Allegro.
2. Chcemy jednorazowo pobrać WSZYSTKIE informacje o jego aukcjach (najlepiej wywołując jak najmniej requestów). Teraz niestety musimy wykonać ponad 2*N requestów (gdzie N to ilość jego aukcji). Najpierw pobieramy listę aukcji za pomocą /sale/offers. Niestety ta metoda nie zwraca wszystkich informacji o aukcji, więc i tak musimy dla każdej z nich wywołać /sale/product-offer (chociażby aby pobrać parametry aukcji). Dodatkowo aby uzyskać informację o Allegro Ceny to dla każdej aukcji musimy wywołać kolejny request. A jak jeszcze chcemy znać klasyfikację SMART to kolejny. Czyli mamy ponad 3*N requestów.
3. Jak już pobierzemy jednorazowo wszystkie oferty - to chcielibyśmy utrzymywać ich aktualny stan. Czyli poprzez dziennik zdarzeń. Problem jest taki, że obecnie zdarzenie CHANGE nie informuje nas o rodzaju zmiany. Więc kolejno dla każdego zdarzenia CHANGE musimy wykonać kilka requestów dla jednej aukcji aby pobrać jej komplet danych.

Jak widać obecne API Allegro jest bardzo nieprzyjazne dla aplikacji obsługującej masowe ilości kont Allegro oraz aukcji.

**Reply by MartaNowaczyk** [reply]
Jeśli jest to kolejna sugestia [dodaj nową](https://github.com/allegro/allegro-api/discussions/new?category=suggestions), jeśli jest to pytanie dotyczące działania API opisz to w nowym wątku w [Issues](https://github.com/allegro/allegro-api/issues).

**Comment by MaciejFrackowiak** [comment]
Zaktualizowaliśmy status sugestii jako "odrzucona", ponieważ nie planujemy sugerowanej zmiany.
